Mainnet/NonFungibleToken/NonFungibleToken/NonFungibleToken.cs

```diff
@@ -0,0 +1,639 @@
+﻿using Stratis.SmartContracts;
+/// <summary>
+/// A non fungible token contract.
+/// </summary>
+public class NonFungibleToken : SmartContract
+{
+    public struct TransferLog
+    {
+        [Index]
+        public Address From;
+        [Index]
+        public Address To;
+        [Index]
+        public ulong TokenId;
+    }
+
+    public struct ApprovalLog
+    {
+        [Index]
+        public Address Owner;
+        [Index]
+        public Address Approved;
+        [Index]
+        public ulong TokenId;
+    }
+
+    public struct ApprovalForAllLog
+    {
+        [Index]
+        public Address Owner;
+        [Index]
+        public Address Operator;
+
+        public bool Approved;
+    }
+
+    public struct OwnershipTransferedLog
+    {
+        [Index]
+        public Address PreviousOwner;
+
+        [Index]
+        public Address NewOwner;
+    }
+
+    /// <summary>
+    /// Get a value indicacting if the interface is supported.
+    /// </summary>
+    /// <param name="interfaceId">The id of the interface to support.</param>
+    /// <returns>A value indicating if the interface is supported.</returns>
+    private bool GetSupportedInterfaces(uint interfaceId)
+    {
+        return this.PersistentState.GetBool($"SupportedInterface:{interfaceId}");
+    }
```

Comment:
This Get method and others below it, should they be private? Would it be beneficial to make these public so external resources can read this data?

Reply:
There was public method for this already and i removed this one.